internal/providerid/providerid.go

```diff
@@ -24,6 +24,20 @@ const (
 	prefixRobotLegacy = "hcloud://bm-"
 )
 
+type UnkownPrefixError struct {
+	ProviderID string
+}
+
+func (e *UnkownPrefixError) Error() string {
+	return fmt.Sprintf(
+		"providerID does not have one of the the expected prefixes (%s, %s, %s): %s",
```

Comment:
```suggestion
		"Provider ID does not have one of the the expected prefixes (%s, %s, %s): %s",
```

Makes it consistent with the rest of the names.